### ISSUE TITLE
map.setMaxBounds(bounds) throws "Failed to invert matrix" for Montana, USA

### ISSUE THREAD

**Issue by fabiancook** [mapbox/mapbox-gl-js#7071]
**mapbox-gl-js version**: v0.47.0

**browser**: Google Chrome Version 67.0.3396.99

### Steps to Trigger Behavior

 1. Use the bounds `[[-111.044275,45.001345], [-111.056207,44.935901]]` as your max bounds

### Link to Demonstration

https://jsbin.com/jayokijuqu/edit?html,css,js,output

### Expected Behavior

The bounds are set 

### Actual Behavior

I get the error "failed to invert matrix"

**Comment by fabiancook** [comment]
I think my issue is something with using the `extend` method of `LngLatBounds`, I have a polygon of Montana and I'm wanting to use that polygon to just show the state, I noticed the longitude of SW & NE are very close which definitely looks like it would cause an issue. 

I have attached the GeoJSON file that I am using for this. 

When applied my mask using the attached file my map looks like:

![image](https://user-images.githubusercontent.com/4337080/43562772-c2b1ae62-9672-11e8-8a43-e8454b0cd0b4.png)

I would expect the following code to _"simply"_ extend the bounds so all values "fit" within. 

```js
    const montanaBounds = Montana.geometry.coordinates.reduce(
      (bounds, coordinates) => {
        return bounds.extend(coordinates);
      },
      new MapboxGL.LngLatBounds()
    );
```


[montana.json.txt](https://github.com/mapbox/mapbox-gl-js/files/2252123/montana.json.txt)
